### ISSUE TITLE
Expose api functions for accessing swagmap data

### ISSUE THREAD

**Issue by limikael** [tunapanda/swag-old#10]
Suggestion:

.../api/swagmaps

Returns an array where each element contains information about a swagmap, minimum data would be id, name and description.

.../api/swagmap/<id>

Returns an array of swagifacts for the swagmap with specified id.

**Comment by usernamenumber** [comment]
This sounds reasonable to me, though I think there's a dependency on us settling on a schema for swagifacts (and I think a further dependency on that is figuring out a standard notation for skills so that swagifacts can say what they provide/require in a consistent way)

**Comment by usernamenumber** [comment]
...I have some thoughts on this ^^^^, and will try to expand them into a proposal while I work on preparing my conference stuff.

**Comment by limikael** [comment]
Awesome! Although it doesn't require that we settle on the schema finally, we can still evolve it... But, yeah, we do need something to start with... So a suggestion would be an array of objects with the following keys:
- **title** <br/> The title of the swagifact
- **description** _optional_<br/> A description of the swagifact. It is up to the swag viewer what it wants to do with this, maybe show it on hover?
- **object** <br/> The xAPI objective url that needs to be completed for this swagifact to show as completed.
- **prerequisite** _optional_<br/> An xAPI objective url that needs to be completed before this swagifact is accessible. This is also what is used to resolve dependencies for the swagifact.
- **url** _optional_<br/> A url specifying a location online where the swagifact can be completed. The suggested behaviour for the swagviewer would be to open up this as the user clicks on the swagifact.

**Comment by usernamenumber** [comment]
Good point about building in the assumption of an evolving spec. Relevant to that: https://github.com/tunapanda/swagmaps/issues/6

**Comment by Jakeii** [comment]
If we're going to stick with using github for the moment, we can just use the github api to get a list of swagmaps, we would need to store them all in the top level for convenience though.

something like `GET https://api.github.com/repos/tunapanda/swagmaps/contents/` could get an index of swagmaps.

Then use rawgit.com to retrive the actual json, `GET https://rawgit.com/tunapanda/swagmaps/master/KTouch/ktouch.json`
